src/fmu/dataio/datastructure/meta/meta.py

```diff
@@ -114,57 +114,72 @@ class User(BaseModel):
     )
 
 
-class FMUCase(BaseModel):
-    name: str = Field(
-        description="The case name",
-        examples=["MyCaseName"],
-    )
-    user: User = Field(
-        description="The user name used in ERT",
-    )
-    uuid: UUID = Field(
-        examples=["15ce3b84-766f-4c93-9050-b154861f9100"],
-    )
-    description: Optional[List[str]] = Field(
-        default=None,
-    )
+class Case(BaseModel):
+    """The ``fmu.case`` block contains information about the case from which this data
+    object was exported.
+
+    .. note:: If an FMU data object is exported outside the case context, this block
+       will not be present.
+    """
+
+    name: str = Field(examples=["MyCaseName"])
+    """The name of the case."""
+
+    user: User
+    """A block holding information about the user."""
+
+    uuid: UUID = Field(examples=["15ce3b84-766f-4c93-9050-b154861f9100"])
+    """The unique identifier of this case. Currently made by fmu.dataio."""
+
+    description: Optional[List[str]] = Field(default=None)
+    """A free-text description of this case."""
 
 
 class Iteration(BaseModel):
-    id: Optional[int] = Field(
-        default=None,
-        description=(
-            "The internal identification of this iteration, e.g. the iteration number"
-        ),
-    )
-    name: str = Field(
-        description="The convential name of this iteration, e.g. iter-0 or pred",
-        examples=["iter-0"],
-    )
-    uuid: UUID = Field(
-        examples=["15ce3b84-766f-4c93-9050-b154861f9100"],
-    )
+    """The ``fmu.iteration`` block contains information about the iteration this data
+    object belongs to."""
+
+    id: Optional[int] = Field(default=None)
+    """The internal identification of this iteration, typically represented by an
+    integer."""
+
+    name: str = Field(examples=["iter-0"])
+    """The name of the iteration. This is typically reflecting the folder name on
+    scratch. In ERT, custom names for iterations are supported, e.g. "pred". For this
+    reason, if logic is implied, the name can be risky to trust - even if it often
+    contains the ID, e.g. "iter-0"
```

Comment:
Perhaps this should be rephrased - it is actually the `id` which is difficult to trust. I can run a case with 3 ensembles named "pred1", "pred2" and "pred3" and they will all be given `fmu.iteration.id: 0` I believe.

Currently all logic is put on `fmu.iteration.name` when consuming data.

Reply:
I just cut it off after the second sentence so it doesn't become stale documentation, but am happy to add additional context if you think it should be there